book-1-orientation/chapters/ADONET_INTRO.md

```diff
@@ -9,624 +9,305 @@ This will be an interactive lesson. Your instructor will lead you through follow
+
+   ```csharp
+   static void Main(string[] args)
+   {
+       var departmentRepo = new DepartmentRepository();
+
+       Console.WriteLine("Getting All Departments:");
+       Console.WriteLine();
+
+       var allDepartments = departmentRepo.GetAllDepartments();
+
+       foreach (var dept in allDepartments)
+       {
+           Console.WriteLine($"{dept.Id} {dept.DeptName}");
+       }
+   }
+   ```
+
+1. Now create another method in `DepartmentRepository` that will get a single department by its Id. The method should accept an `int id` as a paramter and return a single `Department` object.
+
+   ```csharp
+   /// <summary>
+   ///  Returns a single department with the given id.
+   /// </summary>
+   public Department GetDepartmentById(int id)
+   {
+       using (SqlConnection conn = Connection)
+       {
+           conn.Open();
+           using (SqlCommand cmd = conn.CreateCommand())
+           {
+               cmd.CommandText = "SELECT DeptName FROM Department WHERE Id     = @id";
+               cmd.Parameters.Add(new SqlParameter("@id", id));
+               SqlDataReader reader = cmd.ExecuteReader();
+
+               Department department = null;
+
+               // If we only expect a single row back from the database, we    don't need a while loop.
+               if (reader.Read())
+               {
+                   department = new Department
+                   {
+                       Id = id,
+                       DeptName = reader.GetString(reader.GetOrdinal   ("DeptName"))
+                   };
+               }
+
+               reader.Close();
+
+               return department;
+           }
+       }
+   }
+   ```
+
+1. Update `Program.Main` to also call this new method
+
+   ```csharp
+   Console.WriteLine("----------------------------");
+   Console.WriteLine("Getting Department with Id 1");
+
+   var singleDepartment = departmentRepo.GetDepartmentById(1);
+
+   Console.WriteLine($"{singleDepartment.Id} {singleDepartment.DeptName}");
+   ```
+
+1. Now that we've read data from our database, let's look at how we can add new records. Create a new method in the `DepartmentRepository` and name it `AddDepartment`. It should accept a single `Department dept` parameter and not return anything.
+
+   ```csharp
+   /// <summary>
+   ///  Add a new department to the database
+   ///   NOTE: This method sends data to the database,
+   ///   it does not get anything from the database, so there is nothing to    return.
+   /// </summary>
+   public void AddDepartment(Department department)
+   {
+       using (SqlConnection conn = Connection)
+       {
+           conn.Open();
+           using (SqlCommand cmd = conn.CreateCommand())
+           {
+               // These SQL parameters are annoying. Why can't we use  string interpolation?
+               // ... sql injection attacks!!!
+               cmd.CommandText = "INSERT INTO Department (DeptName) Values     (@deptName)";
```

Comment:
I know this wasn't in the original, but I think we should and use ` OUTPUT INSERTED.Id`  here. We can set the id of the added department.

This was an oversight on my part when I wrote this.

Reply:
Remember to switch back to ExecuteReader with an if block too.

Reply:
Why not ExecuteScaler?

Reply:
Ah, I wasn't paying close enough attention.  I usually tell my class to selected `inserted.*` so they can get an entire record back, including default values etc   if you just want the Id execute scalar is perfect.